src/components/Swap/Toolbar/index.tsx

```diff
@@ -61,7 +61,7 @@ function CaptionRow() {
       default:
     }
 
-    if (state === TradeState.LOADING) {
+    if (state === TradeState.LOADING && trade == null) {
       return { caption: <Caption.LoadingTrade gasUseEstimateUSD={gasUseEstimateUSD} /> }
     }
 
```

Comment:
L68 is not the same check as L64 (`trade !== null`), so this will not necessarily always be triggered. Should L68 be changed?

Reply:
no, this is intentional. we only want to show `Caption.LoadingTrade` if it's the first time a trade is loading. after that, we want to show the previously loaded trade while fetching the updated quote.

Reply:
put differently, is there a case where `state === TradeState.LOADING && trade !== undefined && !(inputCurrency && outputCurrency && isAmountPopulated)`, where `Caption.Trade` should be displayed but will not?

It's hard to tell, because you're using two different checks. I think you should use the same check (negated) for these two conditionals.

Reply:
@zzmp so i think it does make sense to add an `if (trade)` check to the `Caption.Trade` case. we do expect that to be defined there (L68 in your original comment). but otherwise the checks should be different